tests/Unit/Numerical/Spectral/Test_MeanValue.cpp

```diff
@@ -0,0 +1,98 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+#include <numeric>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/IndexIterator.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "Numerical/Spectral/LegendreGaussLobatto.hpp"
+#include "Numerical/Spectral/Linearize.hpp"
+#include "Numerical/Spectral/MeanValue.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+// GCC 4.7.3 triggers the following for the CHECK() macros.
+// suggest parentheses around comparison in operand of == [-Wparentheses]
+// adding an extra set of parentheses causes CHECK to print false instead
+// of the lhs and rhs of the assertion when a check fails.
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wparentheses"
+
+SPECTRE_TEST_CASE("Unit.Functors.MeanValue", "[Numerical][Spectral][Unit]") {
+  for (size_t nx = 2; nx < 7; ++nx) {
+    const DataVector& x = Basis::lgl::collocation_points(nx);
+    for (size_t ny = 2; ny < 7; ++ny) {
+      const DataVector& y = Basis::lgl::collocation_points(ny);
+      for (size_t nz = 2; nz < 7; ++nz) {
+        const DataVector& z = Basis::lgl::collocation_points(nz);
+        const Index<3> extents(nx, ny, nz);
+        Scalar<DataVector> u(extents.product());
+        for (IndexIterator<3> i(extents); i; ++i) {
+          u[i.offset()] = exp(x[i()[0]]) * exp(y[i()[1]]) * exp(z[i()[2]]);
+        }
+        DataVector u_lin = linearize(u.get(), extents);
```

Comment:
const, const, const

Reply:
A comment explaining the `u_lin` usage might be nice.. this approach is less intuitive than transforming to spectral space, zeroing out all but the 0'th mode, then transforming back..

Reply:
I have no idea why I wrote this test the way I did (on Bitbucket).  Look at the test for Linearize to see how to do a transform, and then make sure all but the lowest coefficient is zero